src/XProtocol/XProtocol.hh

```diff
@@ -1370,7 +1370,11 @@ static int mapError(int rc)
            case ENOTBLK:       return kXR_NotFile;
            case ENOTSUP:       return kXR_Unsupported;
            case EISDIR:        return kXR_isDirectory;
-           case EEXIST:        return kXR_ItExists;
+           case EEXIST:        [[fallthrough]];
+           // In the case one tries to delete a non-empty directory
+           // we have decided that until the next major release
+           // the kXR_ItExists flag will be returned
+           case ENOTEMPTY:     return kXR_ItExists;
```

Comment:
This is really just nitpicking, so feel free to ignore, but I'd put the `return kXR_ItExists;` on `case EEXIST`, and put `case ENOTEMPTY` as the fall through.

Reply:
Yes, I'll do that :)